desktop/bin/make-changelog.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+CALYPSO_DIR=$(cd $(dirname $0)/../../ && pwd)
+
+# desktop tags
+function get_desktop_tags() {
+  desktop_tags=""
+  declare -i x=0
+  for tag in $(git tag); do
+    # guard against stray v9.9.9 test tag that keeps being re-introduced to the repo! ಠ_ಠ
+    if [[ "$tag" == *"9.9.9" ]]; then
+      continue;
+    fi
+
+    if [[ $tag == v* ]]; then
+      if [ $x -gt 0 ]; then
+        desktop_tags="$desktop_tags"$'\n'"$tag"
+      else
+        desktop_tags="$tag"
+      fi
+    fi
+    x+=1
+  done
+  echo "$desktop_tags"
+}
+
+# fetch all tags in descending lexicographical order
+# (exclude current tag with `awk`)
+tags=$(get_desktop_tags | tr - \~ | sort -V -r | tr \~ - | awk '{if(NR>1)print}')
+
+# get tag for previous stable release from the sorted list
+# (first match without `-`, e.g. v1.2.3, not v1.2.3-alpha1)
+last_stable_tag=$(for tag in $tags; do if [[ ! "$tag" == *"-"* ]]; then
+  echo "$tag"
+  break
+fi; done)
+
+# get the current tag, fall back to HEAD
+current_tag=$VERSION
+if [ -z "$current_tag" ]; then
+  current_tag=HEAD
+fi
+
+# Include commit message (%s). Other elements such as
+# commit author (_%aN_) and commit short hash (%h) can
+# be included if desired.
+git_log_format="%s"
+
+echo "## What's Changed"
+echo ""
+
+# Fill and sort changelog (final sort in commit-date order)
+git_log=$(git log --oneline --pretty=format:"$git_log_format" $last_stable_tag...$current_tag -- "$CALYPSO_DIR/desktop/" "$CALYPSO_DIR/client/lib/desktop" |
+  sort -s -k 1,1)
```

Comment:
Why did you choose to sort this rather than leaving it in the order returned by `git log`?

Reply:
That's a great question. I ported this over from the script in the wp-desktop repository, but I believe the `sort -s -k 1,1` clause ensures that when using semantic commits, `git log` output is sorted both chronologically and by semantic category and sub-category ([link](https://github.com/Automattic/wp-desktop/blob/2fbfbe63e834b6954a891a3bc5ab7686ba6171b5/make-changelog.sh#L35)), i.e:

```
category1: commit n
category1(sub-category 1-1): commit n-1
category1(sub-category 1-1): commit n-2
category2: commit n
category2(sub-category 2-1): commit n-1
// ...
```

This was to ensure that the loop creating the final output was guaranteed that the same category heading could be used until the category name changed between commits ([link](https://github.com/Automattic/wp-desktop/blob/2fbfbe63e834b6954a891a3bc5ab7686ba6171b5/make-changelog.sh#L95-L100)), so we could have nicely-formatted output that printed each category (and sub-category, if relevant) only once before each group of changes:

```
Features

Notifications

- blah blah
- blah blah

Other Stuff

- blah blah

Fixes

// ...

```

Hope that makes sense. Regardless, all that to say ... since we're no longer using semantic commits, this clause is no longer necessary. 😆 Great catch and excluded in 77e2354bed5a5344aa8bb9f2e35b9964f6a650c6.

Reply:
Oh yeah... I can see how it would have been super useful with the semantic commits.